File: Interface Gráfica - Criação de tela/Java/TelaCriacaoHeroi.java
/*
 * Aula 6 P2 - Interface Gráfica com OO: A Tela de Criação de Herói
 *
 * Uso de IA: usei o Claude (Anthropic) para aprender o conceito de
 * JFrame e layout managers do Swing, e ajudar a estruturar da tela.
 * Validei manualmente executando o programa e conferindo a janela e
 * a saída no console.
 */

import javax.swing.*;
import java.awt.*;

public class TelaCriacaoHeroi extends JFrame {

    private JLabel rotuloTitulo;
    private JTextField campoNome;
    private JTextField campoVida;
    private JTextField campoClasse;
    private JButton botaoCriar;

    public TelaCriacaoHeroi() {

        super("IF Quest");
        setSize(400, 300);
        setDefaultCloseOperation(EXIT_ON_CLOSE);

        setLayout(new BorderLayout());

        rotuloTitulo = new JLabel("Novo Herói", SwingConstants.CENTER);
        add(rotuloTitulo, BorderLayout.NORTH);

        JPanel painelFormulario = new JPanel(new GridLayout(3, 2, 5, 5));

        painelFormulario.add(new JLabel("Nome:"));
        campoNome = new JTextField();
        painelFormulario.add(campoNome);

        painelFormulario.add(new JLabel("Vida inicial:"));
        campoVida = new JTextField();
        painelFormulario.add(campoVida);

        painelFormulario.add(new JLabel("Classe (Mago/Guerreiro):"));
        campoClasse = new JTextField();
        painelFormulario.add(campoClasse);

        add(painelFormulario, BorderLayout.CENTER);

        botaoCriar = new JButton("Criar Herói");

        botaoCriar.setEnabled(false);
        add(botaoCriar, BorderLayout.SOUTH);

        rotuloTitulo.setText("Criação de Herói");
        campoNome.setText("Digite o nome aqui");
        System.out.println("Texto atual do campo nome: " + campoNome.getText());
    }

    public static void main(String[] args) {
        TelaCriacaoHeroi tela = new TelaCriacaoHeroi();
        tela.setVisible(true);
    }
}

File: Interface Gráfica - Criação de tela/python/tela_criacao_heroi.py
"""
Aula 6 P2 - Interface Gráfica com OO: A Tela de Criação de Herói (Tkinter)

Uso de IA: usei o Claude (Anthropic) para aprender o conceito de Tk
e os gerenciadores de layout do Tkinter (pack/grid), e ajudar a
estruturar a tela. Validei manualmente executando o programa e
conferindo a janela e a saída no console.
"""

import tkinter as tk

class TelaCriacaoHeroi(tk.Tk):
    def __init__(self):
        
        super().__init__()
        self.title("IF Quest")
        self.geometry("400x300")

       
        self.rotulo_titulo = tk.Label(self, text="Novo Herói", font=("Arial", 14))
        self.rotulo_titulo.pack(side="top", pady=10)

        
        painel_formulario = tk.Frame(self)
        painel_formulario.pack(side="top", pady=10)

        tk.Label(painel_formulario, text="Nome:").grid(row=0, column=0, padx=5, pady=5)
        self.campo_nome = tk.Entry(painel_formulario)
        self.campo_nome.grid(row=0, column=1, padx=5, pady=5)

        tk.Label(painel_formulario, text="Vida inicial:").grid(row=1, column=0, padx=5, pady=5)
        self.campo_vida = tk.Entry(painel_formulario)
        self.campo_vida.grid(row=1, column=1, padx=5, pady=5)

        tk.Label(painel_formulario, text="Classe (Mago/Guerreiro):").grid(row=2, column=0, padx=5, pady=5)
        self.campo_classe = tk.Entry(painel_formulario)
        self.campo_classe.grid(row=2, column=1, padx=5, pady=5)

      .
        self.botao_criar = tk.Button(self, text="Criar Herói", state="disabled")
        self.botao_criar.pack(side="bottom", pady=10)

        self.rotulo_titulo.config(text="Criação de Herói")
        self.campo_nome.insert(0, "Digite o nome aqui")
        print("Texto atual do campo nome:", self.campo_nome.get())


if __name__ == "__main__":
    tela = TelaCriacaoHeroi()
    tela.mainloop()